pkg/storage/chunk/cache/lru_cache.go

```diff
@@ -0,0 +1,249 @@
+package cache
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"sync"
+	"unsafe"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/loki/pkg/logqlmodel/stats"
+	util_log "github.com/grafana/loki/pkg/util/log"
+	lru "github.com/hashicorp/golang-lru/simplelru"
+	"github.com/oklog/ulid"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type codec string
+
+const (
+	codecHeaderSnappy             codec = "dvs" // As in "diff+varint+snappy".
+	codecHeaderSnappyWithMatchers codec = "dm"  // As in "dvs+matchers"
+)
+
+var DefaultLRUCacheConfig = LRUCacheConfig{
+	MaxSizeBytes: "250MB",
+}
```

Comment:
Is this anywhere used?

Reply:
Not really. I think it was one of those parts I took from Mimir/Thanos that were necessary for something that Loki doesn't need.